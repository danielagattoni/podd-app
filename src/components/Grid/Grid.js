import React, { Component } from 'react';
import Card from 'components/Card/Card.js';

class Grid extends Component {

  render() {
    const { podds } = this.props;

    return(
      <div className="Grid">
        {
          podds.map((item, index) => <Card key={index} {...item} />)
        }
      </div>
    );
  }
}

export default Grid;
