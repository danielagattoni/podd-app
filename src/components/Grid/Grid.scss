@import "../../styles/config.mixins.scss";
@import "../../styles/config.variables";

.Grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
  margin-bottom: $gutter-v;
}
