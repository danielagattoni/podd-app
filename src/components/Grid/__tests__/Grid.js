import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Grid from 'components/Grid/Grid.js';

const renderer = new ShallowRenderer();

describe(Grid, () => {

  it('displays Grid component', () => {
    const component = renderer.render(<Grid items={[
      {
        store: {
          id: 1,
          displayName: "Eleven-7",
          address: "410 Collins Street",
          mobile: "0481233123",
          coords: {
            latitude: "-37.816981",
            longitude: "144.9584893"
          },
          rating: 10
        },
        pieOfTheDay: {
          displayName: "Beef",
          quantity: 5,
          priceString: "$3.00",
        }
      },
      {
        store: {
          id: 1,
          displayName: "Brian's Pies",
          address: "366 Lonsdale Street",
          mobile: "0481233123",
          coords: {
            latitude: "-37.816981",
            longitude: "144.9584893"
          },
          rating: 8
        },
        pieOfTheDay: {
          displayName: "Beef and curry",
          quantity: 1,
          priceString: "$2.50",
        }
      }
    ]} />);
    expect(component).toMatchSnapshot();
  });
});
