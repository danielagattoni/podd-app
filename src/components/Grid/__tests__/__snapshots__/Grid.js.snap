// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Grid extends _react.Component {

  render() {const
    items = this.props.items;

    return (
      _react2.default.createElement('div', { className: 'Grid', __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this },

        items.map((item, index) => _react2.default.createElement(_FeaturedCard2.default, { key: index, featured: item, __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: this }))));



  }} displays Grid component 1`] = `
<div
  className="Grid"
>
  <FeaturedCard
    featured={
      Object {
        "pieOfTheDay": Object {
          "displayName": "Beef",
          "priceString": "$3.00",
          "quantity": 5,
        },
        "store": Object {
          "address": "410 Collins Street",
          "coords": Object {
            "latitude": "-37.816981",
            "longitude": "144.9584893",
          },
          "displayName": "Eleven-7",
          "id": 1,
          "mobile": "0481233123",
          "rating": 10,
        },
      }
    }
  />
  <FeaturedCard
    featured={
      Object {
        "pieOfTheDay": Object {
          "displayName": "Beef and curry",
          "priceString": "$2.50",
          "quantity": 1,
        },
        "store": Object {
          "address": "366 Lonsdale Street",
          "coords": Object {
            "latitude": "-37.816981",
            "longitude": "144.9584893",
          },
          "displayName": "Brian's Pies",
          "id": 1,
          "mobile": "0481233123",
          "rating": 8,
        },
      }
    }
  />
</div>
`;
